pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBase.java

```diff
@@ -459,6 +459,12 @@ public void reconsumeLater(Message<?> message, Map<String, String> customPropert
         if (!conf.isRetryEnable()) {
             throw new PulsarClientException(RECONSUME_LATER_ERROR_MSG);
         }
+        if (conf.getSubscriptionType() != SubscriptionType.Shared
+                && conf.getSubscriptionType() != SubscriptionType.Key_Shared) {
+            log.warn("[{}] Reconsume the message {} later for a no-shared subscription {} {} "
+                            + "that will caused message out of order.",
```

Comment:
This log message could be improved. 
```suggestion
            log.warn("[{}] Message {} will be reconsumed without delay. reconsumeLater isn't supported for subscription {} with type {}.",
```

Reply:
>log.warn("[{}] Message {} will be reconsumed without delay. reconsumeLater isn't supported for subscription {} with type {}.",
In fact, we do know the delay time, so saying `be reconsumed without delay` may not be appropriate. Moreover, all subscription types actually support `reconsumeLater`, it's just that it can cause message disorder. We need to alert them to the issue of message disorder. If they are willing, they can still call this API.

Reply:
How about this log
```
log.warn("[{}] Message {} will be reconsumed after delay {} {}. But reconsumeLater isn't recommended for subscription {} with type {}.",

Reply:
> In fact, we do know the delay time, so saying be consumed without delay may not be appropriate. 

Fair enough. It's possible to adjust that in the log message. For example by replacing "without delay" with "without delayed delivery".

`log.warn("[{}] Message {} will be reconsumed without delayed delivery. reconsumeLater isn't supported for subscription {} with type {}.",`

> How about this log
> 
> ```
> log.warn("[{}] Message {} will be reconsumed after delay {} {}. But reconsumeLater isn't recommended for subscription {} with type {}.", 
> ```

I'm sorry I don't understand this message. It doesn't make sense to me.

Reply:
>Fair enough. It's possible to adjust that in the log message. For example by replacing "without delay" with "without delayed delivery".
>`log.warn("[{}] Message {} will be reconsumed without delayed delivery. reconsumeLater isn't supported for subscription {} with type {}.",`

I am sorry, why don't you want to add the delay time in this log?